Assets/Scripts/PerformanceDataManager.cs

```diff
@@ -0,0 +1,449 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+// --- the following are for writing performance files only
+using System.IO;  
+using System.Text; 
+using Unity.MLAgents.Policies; 
+
+public class PerformanceDataManager : MonoBehaviour
+{
+    [SerializeField] private string nnModelName; 
+    [SerializeField] private int numGamesPlayed;  // For agent inference perf tracking
+    private string dataDir;
+    private int startingNumBlocks;
+    private double gameTimeStart = 0;
```

Comment:
It looks gameTimeStart is unused so it and some other functions dedicated to updating it (UpdateGameTimeStart and ResetValues) could be removed. Though maybe talk over Joshua's comment about GameManager's elapsedTime being used in PlayerSupervisor.UpdatePlayerPerformanceData before pulling those gameTimeStart references out.